consensus/hotstuff/committees/leader/leader_selection.go

```diff
@@ -48,7 +48,7 @@ func (l LeaderSelection) FirstView() uint64 {
 }
 
 func (l LeaderSelection) FinalView() uint64 {
-	return l.firstView + uint64(len(l.leaderIndexes))
+	return l.firstView + uint64(len(l.leaderIndexes)) - 1
```

Comment:
This fixes an off-by-one error in the original PR, where we incorrectly extended the `FinalView` one view too far into the future. This is easy to see by considering an epoch with only one view.